src/python/pants/backend/jvm/ivy_utils.py

```diff
@@ -296,8 +296,9 @@ def add_resolved_jars(self, target, resolved_jars):
 
     # Assuming target is a jar library.
     for j in target.jar_dependencies:
-      if j.get_url():
-        self.coordinate_to_attributes[j.coordinate] = {'url': j.get_url()}
+      url = j.get_url(relative=True)
```

Comment:
Done.

Added tests.

Good call for that, I was able find and fix a few things

* relative path should be computed using abspath against (build_root + basepath), `build_root` was missed.
* `normpath` so we get `a/b/c` instead of `/a/b/./c`